packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -547,6 +547,11 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         // Update the used routes in this data store's summarizer node.
         this.summarizerNode.updateUsedRoutes(usedRoutes);
 
+        // If a datastore is referenced again, we will untombstone these routes.
+        if (this.tombstoned) {
```

Comment:
updateUsedRoutes is called for unreferenced data stores as well with an empty array for usedRoutes.
The logic to un-tombstone a data store should be in `dataStores.ts` to keep it consistent with the logic that tombstones it.